### PR TITLE
#1788 Add mobile number column to system imports for supervisors

### LINKED ISSUE #1788: add mobile number column to system imports for supervisors
Part of epic #1017, dependent on #1785

AS an **admin**,
I WANT to be able to import mobile numbers associated with supervisor accounts,
SO THAT I can bulk upload supervisor mobile numbers instead of needing to manually add/edit them one-by-one.

### How it works now
After #1785 is complete, mobile numbers will be able to get added to accounts manually, but not in bulk.

### How it should work
Admins should be able to include mobile numbers in supervisor imports.

### Acceptance criteria
- [x] a new column for mobile number is added to the sample .csv for supervisor system imports
- [x] when mobile numbers get added to existing supervisor accounts via system import, the existing supervisor account is updated accordingly
- [x] if a supervisor account in the system import already has a mobile number associated with it, the existing mobile number is overridden by the new number in the system import

### PR DESCRIPTION
**Pull request by harsohailB**
### What github issue is this PR for, if any?
Resolves #1788 

### What changed, and why?

* `User` data model to store phone number as a `string`
* `supervisor_importer` to include new column for phone numbers and validating the inputs in each row
* Seeding of data to include random generation of phone numbers for users using `Faker`

### How will this affect user permissions?
- Volunteer permissions: N/A
- Supervisor permissions: N/A
- Admin permissions: N/A

### How is this tested? (please write tests!) 💖💪

* Added phone numbers to existing supervisor data csv files
* Created new test csv test files for invalid inputs and missing phone numbers
* Created new tests in the supervisor importer spec file

### Screenshots please :)

**Valid Input:**
Screen Shot 2022-04-02 at 9 01 49 PM

Screen Shot 2022-04-02 at 9 02 15 PM

**Invalid Input:**

* Input
Screen Shot 2022-04-02 at 8 56 51 PM

* Error File
Screen Shot 2022-04-02 at 8 57 13 PM

### CODE REVIEW

File: app/lib/importers/supervisor_importer.rb [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
-      supervisor_params = row.to_hash.slice(:display_name, :email).compact
+      supervisor_params = row.to_hash.slice(:display_name, :email, :phone_number).compact
 
       unless supervisor_params.key?(:email)
         raise "Row does not contain e-mail address."
       end
 
+      if supervisor_params.key?(:phone_number)
+        if supervisor_params[:phone_number].length != VALID_PHONE_NUMBER_LENGTH || !supervisor_params[:phone_number].scan(/\D/).empty?
+          raise "Phone number is not in correct format"
```
Comment (compwron) [inline app/lib/importers/supervisor_importer.rb:23]: Specify correct format in error

**Review by compwron — APPROVED** [review]
Looks good

**Comment by compwron** [comment]
Some strange tests failing, I'm not sure what could have gone wrong there...
```

rspec './spec/datatables/reimbursement_datatable_spec.rb[1:2:5:1:1]' # ReimbursementDatatable multiple record handling order by case number behaves like a sorted results set should order ascending by default
rspec './spec/datatables/reimbursement_datatable_spec.rb[1:2:5:1:3:1]' # ReimbursementDatatable multiple record handling order by case number behaves like a sorted results set descending order should order correctly
rspec './spec/datatables/reimbursement_datatable_spec.rb[1:2:5:1:2:1]' # ReimbursementDatatable multiple record handling order by case number behaves like a sorted results set explicit ascending order should order correctly
```

**Review by xihai01 — COMMENTED** [review]
For your supervisor_import.rb file, I see that you are making validations for the phone number. But would it be better to handle these validations in the User model instead? Since volunteer, supervisor and admin all inherit from the User model and each of these users will need the same validations for phone numbers too. 

https://guides.rubyonrails.org/active_record_validations.html#length

**Comment by harsohailB** [comment]
> But would it be better to handle these validations in the User model instead?

Yes, this is a great point. I will make this change which will also help us bring the `supervisor-importer` function below the limit of 25 lines.

**Comment by harsohailB** [comment]
> Some strange tests failing, I'm not sure what could have gone wrong there...

Yea not really sure what's happening here either as the `rspec / rspec (pull_request` check is passing.

File: app/lib/importers/supervisor_importer.rb (hunk 1 above)
Comment (harsohailB) [reply app/lib/importers/supervisor_importer.rb]: Added as `"Phone number is not in correct format: 1XXXXXXXXXX"`!

**Review by xihai01 — APPROVED** [review]
looks good to me and smart use of custom validators.

**Comment by FireLemons** [comment]
:1st_place_medal:

**Pull request merged**